pkg/controllers/provisioning/provisioner.go

```diff
@@ -136,3 +140,22 @@ func (p *Provisioner) Batch(ctx context.Context) (pods []*v1.Pod) {
 		}
 	}
 }
+
+// RemoveScheduled removes pods that have been assigned a node.
+// This check is needed to prevent duplicate binds when a pod is scheduled to a node
+// between the time it was ingested into the scheduler and the time it is included
+// in a provisioner batch.
+func (p *Provisioner) RemoveScheduled(ctx context.Context, pods []*v1.Pod) []*v1.Pod {
+	unscheduled := []*v1.Pod{}
+	for _, pod := range pods {
+		candidate := pod
+		if err := p.kubeClient.Get(ctx, types.NamespacedName{Namespace: pod.Namespace, Name: pod.Name}, candidate); err != nil {
+			logging.FromContext(ctx).Errorf("Checking if pod \"%s\" is provisionable", pod)
```

Comment:
This error log line seems a bit odd to me. Also, what if the pod is deleted at this point? Perhaps we should continue if errors.IsNotFound, but log an error otherwise? "Unexpected error filtering provisionable pods, %s"

Reply:
tbh I think a lot of our error logging is really weird like this and doesn't read well in the actual log. I've tried to keep with our convention for the most part, but I definitely like explicitly saying "Unexpected error" rather than relying on the log level to convey it's an error. Maybe we should clean up the other error logging.

Reply:
I love this comment! No need to say failed if it's an error!